frontend/src/components/Classroom/ClassroomList.vue

```diff
@@ -0,0 +1,134 @@
+<script setup>
+import { onMounted, ref, watch } from 'vue';
+import classroomService from '@/api/classrooms';
+import DeleteIcon from '@/assets/img/delete_icon.svg';
+import { parseDates } from '@/helpers/parse';
+
+const classrooms = ref(null);
+const searchTerm = ref('');
+const fields = {
+  name: 'Name',
+  capacity: 'Capacity',
+  createdAt: 'Created At',
+};
+
+onMounted(() => filterClassrooms(''));
+watch(searchTerm, newSearchTerm => filterClassrooms(newSearchTerm));
+const filterClassrooms = (search = '') => {
+  classroomService.getAllClassrooms().then(resp => {
+    const parsed = parseDates(resp);
+    const filtered = parsed.filter(room => room.name.toLowerCase().includes(search.toLowerCase()));
+    classrooms.value = filtered;
+  });
+};
+const handleDelete = async classroomId => {
+  const isDeleteConfirmed = confirm('Do you really want to delete classroom?');
+  if (!isDeleteConfirmed) return;
+
+  const deleteResponse = await classroomService.deleteClassroomById(classroomId);
+
+  if ('error' in deleteResponse) {
+    return alert('Internal Server Error. Can not get classroooms.');
+  }
+
+  filterClassrooms(searchTerm.value);
+};
+</script>
+
+<template>
+  <div class="main">
+    <div class="rsprd-bar">
+      <h2 class="rsprd-bar__title">Classrooms</h2>
+      <input v-model="searchTerm" class="rsprd-body__input" type="text" placeholder="Search..." />
```

Comment:
This does not really need to be a `v-model` since something depends on it that is not a computed. In the above comment, I wrote to move it to a computed. If you choose to not move in that direction (or the other one mentioned in the comment) then I advise to remove the `v-model` here and listen to the `input` event directly instead of relying on the watcher to trigger the filter method. Watchers make it more difficult to follow the flow of data. This is a simple use-case so it is not really a problem, but on bigger projects, it can be difficult to debug if an issue arrives.